tests/test_renderers_slate.py

```diff
@@ -0,0 +1,177 @@
+import os
+import pathlib
+import shutil
+
+import pytest
+
+from lablib.lib import SequenceInfo
+from lablib.generators import SlateHtmlGenerator, SlateFillMode
+from lablib.renderers import SlateRenderer
+
+
+SLATE_TEMPLATE_FILE = os.path.join(
+    os.path.dirname(__file__),
+    "..",
+    "templates",
+    "slates",
+    "slate_generic",
+    "slate_generic.html"
+)
+DEFAULT_SEQUENCE_PATH = "resources/public/plateMain/v002"
+DEFAULT_SEQUENCE = SequenceInfo.scan(DEFAULT_SEQUENCE_PATH)[0]
+
+
+def _run_slate_renderer(
+        generator: SlateHtmlGenerator,
+        sequence: SequenceInfo = None,
+        output_path: str = None,
+    ):
+    """ Run a renderer associated with the provided
+        generator and source sequence.
+    """
+    sequence = sequence or DEFAULT_SEQUENCE
+    renderer = SlateRenderer(
+        generator,
+        sequence,
+        destination=output_path,
+    )
+
+    renderer.render()
+
+
+@pytest.fixture()
+def source_dir(test_staging_dir, request):
+    """ Prepare are clean source sequence
+        and after each tests.
+    """
+    # Prepare a temporary directory with a copy of default sequence
+    temp_dir = os.path.join(
+        test_staging_dir,
+        request.node.name
+    )
+    os.mkdir(temp_dir)
```

Comment:
```suggestion
    temp_dir = test_staging_dir / request.node.name
    temp_dir.mkdir(exist_ok=True)
```

Reply:
This way we can use incoming Path object and also make sure it is not complaining about existing folder.

Reply:
Actualy the folder had to be removed first since the previous test results were manipulating testing process